backend/test/database.cpp

```diff
@@ -0,0 +1,102 @@
+#define CATCH_CONFIG_MAIN
+#include <catch2/catch_test_macros.hpp>
+#include <fstream>
+#include "database.hpp"
+
+std::string path = "test.db";
+std::string failpath = "random.db";
+std::ofstream file(path);
```

Comment:
**question**: Why can't this be in the top level of the `TEST_CASE`? It's kind of an anti-pattern to use global variables with Catch2. You lose all the benefits of the macro system. Could you have a teardown at the end of the `TEST_CASE` like before? If you want to same file across each section, then you could use nested sections and then teardown after the outer section. Teardown should not be a `SECTION`. It's not testing the database, it's cleaning up the test.